igvc_navigation/src/mapper/main.cpp

```diff
@@ -95,78 +136,46 @@ void frame_callback(const pcl::PointCloud<pcl::PointXYZ>::ConstPtr &msg, const s
     {
       ROS_ERROR_STREAM("\n\nfailed to find transform using empty transform\n\n");
       // TODO: Remove this, this node shouldn't work if there's no transform.
-      tf::StampedTransform transform;
-      transform.setOrigin(tf::Vector3(0.0, 0.0, 0.0));
-      tf::Quaternion q;
-      q.setRPY(0, 0, 0);
-      transform.setRotation(q);
-      transform.child_frame_id_ = "/base_footprint";
-      transform.frame_id_ = "/lidar";
-      transform.stamp_ = ros::Time::now();
-      transforms.insert(std::pair<std::string, tf::StampedTransform>(topic, transform));
     }
   }
-  // Apply transformation to msg points using the transform for this topic.
-  pcl_ros::transformPointCloud(*msg, *transformed, transforms.at(topic));
+}
 
-  // TODO: Change point to point_iter or point_it.
-  pcl::PointCloud<pcl::PointXYZ>::const_iterator point;
 
-  for (point = transformed->begin(); point < transformed->points.end(); point++)
-  {
-    double x_loc, y_loc;
-    std::tie(x_loc, y_loc) = rotate(point->x, point->y);
+void frame_callback(const pcl::PointCloud<pcl::PointXYZ>::ConstPtr &msg, const std::string &topic)
+{
+  // transform pointcloud into the occupancy grid, no filtering right now
+
+  // make transformed clouds
+  pcl::PointCloud<pcl::PointXYZ>::Ptr transformed =
+      pcl::PointCloud<pcl::PointXYZ>::Ptr(new pcl::PointCloud<pcl::PointXYZ>());
+
+  // TODO: time should be based on message timestamp.
+  ros::Time time = ros::Time::now();
+  // Check if static transform already exists for this topic.
+  checkExistsStaticTransform(msg, topic);
+
+  // Apply transformation to msg points using the transform for this topic.
+  pcl_ros::transformPointCloud(*msg, *transformed, transforms.at(topic));
+  handlePointCloud(transformed);
 
-    // TODO: Clear up these variable names. x_point, x_robot, etc.
-    int point_x = static_cast<int>(std::round(x_loc / resolution + cur_x / resolution + start_x));
-    int point_y = static_cast<int>(std::round(y_loc / resolution + cur_y / resolution + start_y));
-    if (point_x >= 0 && point_y >= 0 && point_x < length_y && start_y < width_x)
-    {
-      // TODO: NO MAGIC NUMBERS.
-      // TODO: Have variables for probability thresholds/steps.
-      // TODO: Investigate decay, potentially use CV_16U, decrement at a certain rate.
-      // TODO: Potentially change
-      if (published_map->at<uchar>(point_x, point_y) < 230)
-      {
-        published_map->at<uchar>(point_x, point_y) += (uchar)125;
-      }
-      count++;
-    }
-    else if (!offMap)
-    {
-      ROS_WARN_STREAM("Some points out of range, won't be put on map.");
-      offMap = true;
-    }
-  }
   img_bridge = cv_bridge::CvImage(msgBoi.header, sensor_msgs::image_encodings::MONO8, *published_map);
```

Comment:
change to new image size

Reply:
Can you clarify what you mean by new image size?

Reply:
CV16U vs CV8U, i.e. if we get a byte or two bytes to represent probability.

Reply:
So are we switching to CV16U instead of CV8U?